src/Interfaces.ts

```diff
@@ -1,72 +1,125 @@
-export interface Claim<T extends ClaimAttributes = ClaimAttributes> {
+import * as Joi from 'joi'
+import * as jsigs from 'jsonld-signatures'
+
+export enum ClaimType {
+  Identity = 'Identity',
+  Work = 'Work',
+}
+
+export interface ClaimAttributes {
+  readonly [key: string]: unknown
+}
+
+export interface Claim {
   readonly '@context'?: any
   readonly id?: string
-
-  readonly issuer?: string
-  readonly issuanceDate?: Date // should replace dateCreated... or created... once we remove protobufs
-  readonly publicKey?: string
-  readonly signature?: string
-  readonly created?: string
+  readonly issuer: string
+  readonly issuanceDate: string
   readonly type: ClaimType
-  readonly attributes: T // TODO: replace attributes with claim, once we remove protobufs
+  readonly 'https://w3id.org/security#proof'?: any
+  readonly claim: ClaimAttributes
 }
 
-interface ClaimContext {
+export interface ClaimContext {
   readonly '@context': {
-    readonly [key: string]: string
+    readonly [key: string]: unknown
   }
 }
 
-export function isClaim(object: any): object is Claim {
-  // TODO: use joi or protobuf
-  return !!(object.id && object.publicKey && object.signature && object.type && object.attributes && object.created)
+export interface ClaimTypeContexts {
+  readonly [key: string]: ClaimContext
 }
 
 // WARNING: This MUST account for ALL of the attributes in a Claim, except for id, @context, and signature.
 // Otherwise those attributes without context will be left out of the canonized/signed claim.
 // Refer to https://www.w3.org/2018/jsonld-cg-reports/json-ld/#the-context
-export const ClaimContext: ClaimContext = {
+export const DefaultClaimContext: ClaimContext = {
   '@context': {
-    attributes: 'http://schema.org/CreativeWork',
-    author: 'http://schema.org/author',
-    text: 'http://schema.org/text',
-    created: 'http://purl.org/dc/terms/created',
-    dateCreated: 'http://schema.org/dateCreated',
-    datePublished: 'http://schema.org/datePublished',
-    name: 'http://schema.org/name',
-    tags: 'http://schema.org/keywords',
+    cred: 'https://w3id.org/credentials#',
+    schema: 'http://schema.org/',
+
+    issuer: 'cred:issuer',
+    issuanceDate: 'cred:issued',
     type: 'http://schema.org/additionalType',
-    publicKey: 'http://schema.org/Text',
+    claim: 'http://schema.org/Thing', // The most generic definition in schema.org
   },
 }
 
-export interface ClaimAttributes {
-  readonly [key: string]: string
+export const DefaultWorkClaimContext: ClaimContext = {
+  '@context': {
+    schema: 'http://schema.org/',
+
+    author: 'schema:author',
+    claim: 'http://schema.org/CreativeWork',
```

Comment:
should be https maybe? I went to the link and it redirected to https

Reply:
No, the canonical look up is http. But the website is https.
> Documentation Schemas Home
> CreativeWork
> Canonical URL: http://schema.org/CreativeWork